### ISSUE TITLE
Normalize token values used for composite parameters per 1366 schema design

### ISSUE THREAD

**Issue by punktilious** [IBM/FHIR#1683]
Currently tokens associated with composite parameters use the legacy design, leading to less-than-ideal storage efficiency, and performance impacts for ingestion from the extra indexes on the row_id column.

Migrate composites to use the normalized schema design from issue #1366 (common_token_values and TOKEN_VALUES_V) . This requires some careful planning, because additions to common_token_values must be combined with the inserts required for the non-composite parameters so that they can be sorted prior to insertion to avoid deadlock issues when processing bundles in parallel.

The new design must collect all the composite values and attach to the transaction-data object (see new token-value storage from 1366). This will also greatly improve performance by reducing the number of individual database operations.

**Comment by lmsurpre** [comment]
Stretch goal for 4.6.0

**Comment by d0roppe** [comment]
Ran tests in general performance and also specific composit searches.
